core/src/main/java/software/amazon/awssdk/core/util/Mimetypes.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringTokenizer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.utils.IoUtils;
+
+/**
+ * Utility class that maintains a listing of known Mimetypes, and determines the
+ * mimetype of files based on file extensions.
+ * <p>
+ * This class is obtained with the {#link {@link #getInstance()} method that
+ * recognizes loaded mime types from the file <code>mime.types</code> if this
+ * file is available at the root of the classpath. The mime.types file format,
+ * and most of the content, is taken from the Apache HTTP server's mime.types
```

Comment:
Should we double check to make sure there haven't been any changes in Apache's that we want to include?